src/leiningen/v.clj

```diff
@@ -103,15 +103,34 @@
   (let [v (str (or (version (:v project)) "UNKNOWN"))
         vk (str (or (:manifest-version-name (:v project)) "Implementation-Version"))]
     (-> project
-       (assoc-in [:version] v)
-       (assoc-in [:manifest vk] v))))
+        (assoc-in [:version] v)
+        (assoc-in [:manifest vk] v))))
+
+(defn- update-dependency [v d]
+  (if-not (second d)
+    (assoc d 1 v)
+    d))
+
+; => (update-dependency "42" ["abc" "1.0"])
+; ["abc" "1.0"]
+; => (update-dependency "42" ["abc" (quote lein-v)])
+; ["abc" "42"]
+
+
+(defn dependency-version-from-scm
+  [project]
+  (let [v (str (or (version (:v project)) "UNKNOWN"))
+        vk (str (or (:manifest-version-name (:v project)) "Implementation-Version"))]
+    (update-in project
+               [:dependencies]
+               #(map (partial update-dependency v) %1))))
 
 (defn add-workspace-data
   [project]
   (if-let [wss (workspace-state project)]
     (-> project
-        (assoc-in ,, [:workspace] wss)
-        (assoc-in ,, [:manifest "Workspace-Description"] (:describe wss))
-        (assoc-in ,, [:manifest "Workspace-Tracking-Status"] (string/join " || " (get-in wss [:status :tracking])))
-        (assoc-in ,, [:manifest "Workspace-File-Status"] (string/join " || " (get-in wss [:status :files]))))
+        (assoc-in,, [:workspace] wss)
+        (assoc-in,, [:manifest "Workspace-Description"] (:describe wss))
+        (assoc-in,, [:manifest "Workspace-Tracking-Status"] (string/join " || " (get-in wss [:status :tracking])))
+        (assoc-in,, [:manifest "Workspace-File-Status"] (string/join " || " (get-in wss [:status :files]))))
     project))
```

Comment:
I really like this PR -it turns out that I have had the same problem in a monorepo and have always hacked around with foreknowledge of where the repo would be at the time of a near-future release.  This is *much* cleaner and your implementation is spot-on. Thank you.

Please let me know if you have any questions about my comments and I look forward to a merge.